### ISSUE TITLE
[Bug]: getEvents() lost record data

### ISSUE THREAD

**Issue by brunocascio** [GuavaCZ/calendar#27]
### What happened?

I'm embbeding the calendar widget within a custom filament page. Everything works ok, until a change is performed which dispatches an update, hence a re-call to `getEvents()` and a failure about "record is null"

When I add or delete I lost my `$this->record` previously passed as an argument and working in the initial request

### How to reproduce the bug

```php
class ServiceAvailabilitiesWidget extends CalendarWidget
{
...
  public function getEvents(array $fetchInfo = []): array
  {
          return ServiceAvailability::query()
              ->where('service_id', $this->getRecord()->getKey()) <-- here to filter
              ->get()
              ->map(fn (ServiceAvailability $availability) => $availability->toEvent())
              ->toArray();
  }
...
}
```

Crea a filament Page

```php
<?php

namespace App\Filament\Resources\ServiceResource\Pages;

use App\Filament\Resources\ServiceResource;
use Filament\Resources\Pages\Concerns\InteractsWithRecord;
use Filament\Resources\Pages\Page;

class ServiceAvailabilities extends Page
{
    use InteractsWithRecord;

    protected static string $resource = ServiceResource::class;

    protected static string $view = 'filament.resources.service-resource.pages.service-availabilities';

    public function mount(int | string $record): void
    {
        $this->record = $this->resolveRecord($record);
    }
}
```

The blade view:

```php
<x-filament-panels::page>
  @livewire(App\Filament\Widgets\ServiceAvailabilitiesWidget::class, ['record' => $record])
</x-filament-panels::page>
```

### Package Version

1.4.0

### PHP Version

8.3.x

### Laravel Version

11.9.x

### Which operating systems does with happen with?

macOS

**Comment by lukas-frey** [comment]
The "record" inside the calendar widget has nothing to do with your filament page record.

Inside the calendar widget, record is most of the times "null" and is only set when an action is mounted or when the context menu is opened, for example.

**Comment by lukas-frey** [comment]
Should be now fixed in [v1.4.1](https://github.com/GuavaCZ/calendar/releases/tag/1.4.1). 

`getRecord()` and `getModel()` is now deprecated for backwards compatibility and you should use `getEventRecord()` and `getEventModel()` respectively.

The property is now called `eventRecord`.

So whenever you work with the record of the event from within the calendar widget, you should use `getEventRecord()`.

If you use a resource widget and need to access the current resource record, you need to add a public `$record` property to your widget and access that, as described in the [filament documentation here](https://filamentphp.com/docs/2.x/admin/resources/widgets#accessing-the-current-record).


If you still encounter any issues, please provide a reproduction repository with what you're trying to achieve, so I can test and debug it.

**Comment by brunocascio** [comment]
It works like a charm! thank you!
